main.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// main is the entrypoint for the action.
+func main() {
+	// TODO - This is a port of the entrypoint.sh script.
+	// This is still a work in progress.
+	if err := initalizeENVVariables(); err != nil {
+		panic(err)
+	}
+}
+
+// initalizeENVVariables is a function to initialize the environment variables required for the action.
+//nolint
+func initalizeENVVariables() error {
+	/*
+	 https://docs.github.com/en/actions/learn-github-actions/environment-variables
+	   GITHUB_EVENT_PATH contains the json file for the event.
+	   GITHUB_SHA contains the commit hash.
+	   GITHUB_WORKSPACE contains the repo folder.
+	   GITHUB_EVENT_NAME contains the event name.
+	   GITHUB_ACTIONS is true in GitHub env.
+	*/
+	if err := os.Setenv("ENABLE_SARIF", "1"); err != nil {
```

Comment:
wan to create a set of errors we can re-use?

Reply:
Can I do it as part of the next PR?

Reply:
sure!

Reply:
+1. Will be useful to have re-usable fns here. Maybe consider a `map[string]string` which can be iterated over to set the envVars. Can do this in next PR.

Reply:
Sounds good. Thanks, guys!